File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN;

        int currSum = 0;
        for(int i=0; i<nums.size(); i+=1){
            currSum = max(currSum + nums[i], nums[i]);

            maxSum = max(maxSum, currSum);

            if (currSum < 0){
                currSum = 0;
            }
        }

        return maxSum;
    }
};

File: 53-maximum-subarray/maximum-subarray.go
func maxSubArray(nums []int) int {
    max_sum := math.MinInt
    sum := 0
    for i := range nums {
        sum = max(sum + nums[i], nums[i])

        max_sum = max(max_sum, sum)

        if sum < 0 { sum = 0 }
    }

    return max_sum
}